client/handle_http.go

```diff
@@ -2381,6 +2421,94 @@ func (te *TransferEngine) walkDirUpload(job *clientTransferJob, transfers []tran
 	return err
 }
 
+// This function performs the ls command by walking through the specified directory and printing the contents of the files
```

Comment:
Don't we also have a function that does nearly the same thing -- this is basically the walkDavDir without recursing.  Let's avoid the duplication.

Reply:
I think the current changes made the functions a decent amount different from each other that this is not needed anymore. However, let me know if you still think otherwise.